src/PointwiseFunctions/Hydro/QuadrupoleFormula.hpp

```diff
@@ -0,0 +1,109 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/TagsDeclarations.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/Hydro/TagsDeclarations.hpp"  // IWYU pragma: keep
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace gsl {
+template <typename>
+struct not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace hydro {
+
+namespace Tags {
+/// Tag containing the quadrupole moment, using
+/// TildeD\f$* x^ix^j\f$ (equation 21 of \cite Shibata2003),
+/// with \f$x\f$ the coordinates in Frame Fr.
+template <typename DataType, size_t Dim, typename Fr = Frame::Inertial>
+struct QuadrupoleMoment : db::SimpleTag {
+  using type = tnsr::ii<DataType, Dim, Fr>;
+};
+
+/// Tag containing the first derivative of the quadrupole moment, using
+/// TildeD\f$* (v^ix^j + x^iv^j)\f$ (equation 23 of \cite Shibata2003), with
+/// \f$x\f$ the coordinates in Frame Fr and \f$v\f$ the corresponding velocity.
+template <typename DataType, size_t Dim, typename Fr = Frame::Inertial>
+struct QuadrupoleMomentDerivative : db::SimpleTag {
+  using type = tnsr::ii<DataType, Dim, Fr>;
+};
+}  // namespace Tags
+
+/// Computes the quadrupole moment, using
+/// tilde_d\f$* x^ix^j\f$ (equation 21 of \cite Shibata2003), with \f$x\f$
+/// the coordinates in Frame Fr. Result of calculation stored in result.
+template <typename DataType, size_t Dim, typename Fr = Frame::Inertial>
+void quadrupole_moment(
+    const gsl::not_null<tnsr::ii<DataType, Dim, Fr>*> result,
+    const Scalar<DataType>& tilde_d,
+    const tnsr::I<DataType, Dim, Fr>& spatial_position);
+
+/// Computes the first derivative of the quadrupole moment, using
+/// tilde_d\f$* (v^ix^j + x^iv^j)\f$ (equation 23 of \cite Shibata2003),
+/// with \f$x\f$ the coordinates in Frame Fr and \f$v\f$ the corresponding
+/// velocity. Result of calculation stored in result.
+template <typename DataType, size_t Dim, typename Fr = Frame::Inertial>
+void quadrupole_moment_derivative(
+    const gsl::not_null<tnsr::ii<DataType, Dim, Fr>*> result,
+    const Scalar<DataType>& tilde_d,
+    const tnsr::I<DataType, Dim, Fr>& spatial_position,
+    const tnsr::I<DataType, Dim, Fr>& spatial_velocity);
+
+namespace Tags {
```

Comment:
I advised you poorly here. To avoid cyclic dependencies, the compute tags should get moved into a file in `src/Evolution/Systems/GrMhd/ValenciaDivClean/` maybe just a `QuadrupoleFormula.hpp` file there. We've been unhappy about system tags being in `Evolution` but sometimes needing them in `PointwiseFunctions` but just haven't gotten around to that giant refactor...